src/responder/nss/nss_protocol_grent.c

```diff
@@ -349,6 +350,19 @@ nss_protocol_fill_initgr(struct nss_ctx *nss_ctx,
                                                     SYSDB_PRIMARY_GROUP_GIDNUM,
                                                     0);
 
+    /* Try to get the real gid in case the primary group's gid was overriden. */
```

Comment:
Just a minor nitpick about the comment, I would recommend to say "... in case the original primary group's gid ..." to avoid confusion with all those different primary groups.